pkg/common/observability/otel_observability_service.go

```diff
@@ -0,0 +1,149 @@
+package observability
+
+import (
+	"context"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/observability"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	// TODO: What should we put here?
+	instrumentationName = "github.com/keptn/go-utils/observability/cloudevents"
+)
+
+type OTelObservabilityService struct {
+	TracerProvider trace.TracerProvider
+	Tracer         trace.Tracer
+}
+
+// When creating the cloudevent HTTP Client, by passing the option 'WithObservabilityService', the context
+// gets enriched with the incoming parent trace, thus enabling context propagation
+func (n OTelObservabilityService) InboundContextDecorators() []func(context.Context, binding.Message) context.Context {
```

Comment:
```suggestion
func (os OTelObservabilityService) InboundContextDecorators() []func(context.Context, binding.Message) context.Context {
```
These are usually shortened to the first letters of the class they are representing, e.g. `os` or `oos` in this case.

Reply:
Good catch. Modified all of them to use `os` as it's shorter.